src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -837,10 +931,93 @@ private int decodeSession(
         }
     }
 
+    private final class Subscription
+    {
+        private final MqttServer server;
+        private final int fullReasonCodesMask;
+        private final List<Byte> reasonCodes;
+
+        private int reasonCodesMask;
+
+        private Subscription(
+            MqttServer server,
+            int reasonCodesCount)
+        {
+            this.server = server;
+            this.reasonCodes = new ArrayList<>();
+            for (int i = 0; i < reasonCodesCount; i++)
+            {
+                reasonCodes.add((byte) 0x8f);
+            }
+            this.fullReasonCodesMask = 1 << (reasonCodesCount - 1);
```

Comment:
Let's change this to 2 masks; one `ackMask` to say which subscribes have acked, and one `successMask` to say which subscribes were successful.

The initial value of the `ackMask` should match `unrouteableMask` from `onMqttSubscribe` since those ones are complete, and unsuccessful. The initial value of `successMask` is `0`.

The last pending subscribe has been acked when the bit count of `ackMask` reaches `reasonCodesCount`, which we can probably rename to `ackCount`.